packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -407,29 +459,64 @@ async function createNoteFromTemplate(
     resolvedValues,
     templateMetadata
   );
-  const defaultFilename = path.basename(defaultFilepath.path);
 
   const filepath = await askUserForFilepathConfirmation(
-    defaultFilepath,
-    defaultFilename
+    defaultFilepath.fsPath,
+    path.basename(defaultFilepath.path)
   );
 
   if (filepath === undefined) {
     return;
   }
-  const filepathURI = Uri.file(filepath);
-  await writeTemplate(templateSnippet, filepathURI);
+  const filepathUri = Uri.file(filepath);
+  await writeTemplate(templateSnippet, filepathUri);
+}
+
+function askTemplateCreationLocation(): Thenable<QuickPickItem | undefined> {
+  const homeDirQuickPick = {
+    label: 'In your home directory',
+    description: 'Available for all workspaces',
+    detail: homeDirTemplatesDir.fsPath,
+  };
+
+  if (workspace.workspaceFolders) {
+    const workspaceQuickPick = {
+      label: 'In this workspace',
+      description: 'Available for this workspace',
+      detail: Uri.joinPath(
+        workspace.workspaceFolders[0].uri,
+        '.foam',
+        'templates'
+      ).fsPath,
+    };
+
+    const quickPicks = [workspaceQuickPick, homeDirQuickPick];
+    return window.showQuickPick(quickPicks, {
+      placeHolder: 'Where should the new template be created?',
+    });
+  } else {
+    return Promise.resolve(homeDirQuickPick);
+  }
 }
 
 async function createNewTemplate(): Promise<void> {
   const defaultFilename = 'new-template.md';
-  const defaultTemplate = Uri.joinPath(templatesDir, defaultFilename);
+  const templateLocationSelection = await askTemplateCreationLocation();
+  if (templateLocationSelection === undefined) {
+    return;
+  }
+
+  const defaultFilepath = Uri.joinPath(
+    Uri.file((templateLocationSelection as QuickPickItem).detail),
+    defaultFilename
+  ).fsPath;
+  const defaultFilepathNoExtension = defaultFilepath.replace(/\.[^.]+$/, '');
```

Comment:
I have seen this in a couple of places, plus I think it could be useful in general.. how about moving it to `URI` (in `foam-core`?